src/main/java/wolox/training/models/Book.java

```diff
@@ -78,14 +79,16 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
-        genre = genre;
+        Preconditions.checkNotNull(genre);
```

Comment:
You are validating that the attribute is not null, but you are never validating that if you need a string, come a string and not a string empty, for this validation you can use the following line:
Preconditions.checkArgument(!genre.isEmpty(), "Mensaje Error");
Other good practice is that you have a file with all errors of your app, for this, you should create a file (class), for example "ErrorConstans" in the folder "utils". In this file you should create your constants, for example:
public static String NOT_NULL_GENRE = "The genre field cannot be null.";
then your parameter check line is:
Preconditions.checkArgument(!genre.isEmpty(), ErrorConstants.NOT_NULL_GENRE);

Reply:
that is perfect example, I made it this way.